src/API/TransactionName.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Solarwinds\ApmPhp\API;
+
+use OpenTelemetry\API\Behavior\LogsMessagesTrait;
+use OpenTelemetry\API\Trace\LocalRootSpan;
+use OpenTelemetry\SDK\Trace\ReadableSpanInterface;
+use Solarwinds\ApmPhp\Trace\SpanProcessor\TransactionNameSpanProcessor;
+
+final class TransactionName
+{
+    use LogsMessagesTrait;
+
+    /**
+     * Retrieve the local root span (root-most active span which has a remote or invalid parent) and try to set transaction name in it
+     */
+    public static function set(string $name): bool
+    {
+        $span = LocalRootSpan::current();
+        if ($span instanceof ReadableSpanInterface) {
+            $parentSpanContext = $span->getParentContext();
+            if ($parentSpanContext->isValid() && !$parentSpanContext->isRemote()) {
+                return false;
```

Comment:
Could we have some debug logs for customer before each `return` with reason for failure, or notification of success?

e.g for this one it could be

```suggestion
                $this->logDebug('Cannot set custom transaction name because entry span not started');
                return false;
```

Reply:
I removed that checking because LocalRootSpan::current will actually return the root most active service entry span.

Reply:
Done